src/terra-dev-site/doc/application/reference.1/components/ApplicationBase.doc.mdx

```diff
@@ -40,4 +40,8 @@ ApplicationBase renders an [ApplicationLoadingOverlayProvider](/components/terra
 
 ### Pending Actions
 
-ApplicationBase monitors the application for the presence of rendered [NavigationPrompts](/components/terra-application/application/reference/components/navigation-prompt). ApplicationBase will ensure that the user is prompted prior to closing the application if any [NavigationPrompts](/components/terra-application/application/reference/components/navigation-prompt) are rendered at the time of the unload request.
+ApplicationBase monitors its children for the presence of rendered [NavigationPrompts](/components/terra-application/application/reference/components/navigation-prompt). ApplicationBase will ensure that the user is prompted using a browser-native confirmation dialog prior to any window unload event if any [NavigationPrompts](/components/terra-application/application/reference/components/navigation-prompt) are rendered at the time of the unload request.
+
+### Code Splitting
+
+ApplicationNavigation renders a [Suspense](https://reactjs.org/docs/code-splitting.html#reactlazy) component around its children. If any child components or their descendants are lazy loaded using [React.lazy](https://reactjs.org/docs/code-splitting.html#reactlazy), a sensible fallback will be rendered while the component is being retrieved. The fallback will render below the ApplicationNavigation header and allow the header to remain interactable.
```

Comment:
> any child components or their descendants

Really this will render a loading indicator on the full tree if React.Lazy is used at a way lower level?

Reply:
It would indeed. The alternative would be nothing rendering at all until that script is retrieved.